x/gamm/types/pool.go

```diff
@@ -7,16 +7,18 @@ import (
 	"strings"
 	"time"
 
-	"github.com/tendermint/tendermint/crypto"
+	"github.com/c-osmosis/osmosis/v043_temp/address"
+	proto "github.com/gogo/protobuf/proto"
 
-	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 )
 
-// PoolAccountI defines an account interface for pools that hold tokens.
-type PoolAccountI interface {
-	authtypes.AccountI
+// PoolI defines an interface for pools that hold tokens.
+type PoolI interface {
+	proto.Message
```

Comment:
Why does it implement the proto.Message interface?

Reply:
All the interfaces in the SDK are supposed to implement proto.Message

Reply:
Examples:

https://github.com/cosmos/cosmos-sdk/blob/9fd866e3820b3510010ae172b682d71594cd8c14/crypto/types/types.go#L10

https://github.com/cosmos/cosmos-sdk/blob/9fd866e3820b3510010ae172b682d71594cd8c14/crypto/types/types.go#L10

https://github.com/cosmos/cosmos-sdk/blob/9fd866e3820b3510010ae172b682d71594cd8c14/x/evidence/exported/evidence.go#L12

Reply:
ah I see. It literally just implements clone